### PR TITLE
Update api module docstring for chaco.api

### PR DESCRIPTION
**Pull request by aaronayres35**
Follow up to #649

This PR populates and move the colormap section to the bottom of the module (although this move may have been unnecessary as I had failed to notice the __all__ previously)
It also moves  `BandedMapper` into the mappers section and `ColormappedSelectionOverlay` into the visual components section

### CODE REVIEW

**Comment by aaronayres35** [comment]
Seeing this test failure on CI:
```
======================================================================
ERROR: chaco.tests.test_colormapped_scatterplot (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: chaco.tests.test_colormapped_scatterplot
Traceback (most recent call last):
  File "/home/runner/.edm/envs/chaco-test-3.6-null/lib/python3.6/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/home/runner/.edm/envs/chaco-test-3.6-null/lib/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/home/runner/.edm/envs/chaco-test-3.6-null/lib/python3.6/site-packages/chaco/tests/test_colormapped_scatterplot.py", line 7, in <module>
    from chaco.api import (
ImportError: cannot import name 'jet'

----------------------------------------------------------------------
```
I am not sure why this didn't happen before as we were doing `import *` from the module which defines `__all__` and `jet` was not in `__all__`

File: chaco/api.py [hunk 1]
```diff
@@ -306,6 +384,91 @@
...
 from .abstract_controller import AbstractController
 
+# Colormaps and color palettes
+from .default_colormaps import (
+    center,
+    color_map_dict,
+    color_map_functions,
+    color_map_name_dict,
+    reverse,
+    jet,
```
Comment (rahulporuri) [inline chaco/api.py:394]: i'm sorry for not being clear. IIRC, `jet` isnt the first color function defined in the `default_colormaps` module. Can you check the order of functions vs the list at the end of the module in `default_colormaps` - and fix it if necessary - and then update this import statement to reflect the same changes.

Comment (aaronayres35) [reply chaco/api.py:394]: Yep will do

Comment (aaronayres35) [reply chaco/api.py:394]: Even further, there are color maps that don't make it into the list / into __all__
Should I add them?  It is unclear if that was intentional or accidental.  For example `binary`

Comment (rahulporuri) [reply chaco/api.py]: definitely add them.

Comment (aaronayres35) [reply chaco/api.py]: Sounds good, thats what I ended up doing in the last commit, and it looks like it was just `binary`

File: chaco/api.py [hunk 2]
```diff
@@ -161,6 +154,95 @@
 
 - :class:`~.AbstractController`
 
+Colormaps and Color Palettes
+----------------------------
+
```
Comment (rahulporuri) [inline chaco/api.py:159]: I would prefer adding a subsection here for color maps and another for color palettes

**Review by rahulporuri — APPROVED** [review]
LGTM with a small comment.

File: chaco/api.py (hunk 2 above)
Comment (aaronayres35) [reply chaco/api.py:159]: I also added a "Utilities / Convenience Objects" subsection here as well

**Review by rahulporuri — APPROVED** [review]
Still LGTM

**Comment by aaronayres35** [comment]
This test failure has returned... 
```
======================================================================
ERROR: test_dont_crash_on_click (chaco.tests.test_plot.TestEmptyPlot)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\chaco\chaco\.edm\envs\chaco-test-3.6-pyqt\lib\site-packages\chaco\tests\test_plot.py", line 134, in test_dont_crash_on_click
    [(1, 1), (2, 2), (3, 3), (4, 4)],
  File "D:\a\chaco\chaco\.edm\envs\chaco-test-3.6-pyqt\lib\contextlib.py", line 88, in __exit__
    next(self.gen)
  File "D:\a\chaco\chaco\.edm\envs\chaco-test-3.6-pyqt\lib\site-packages\traitsui\testing\tester\ui_tester.py", line 106, in create_ui
    process_cascade_events()
  File "D:\a\chaco\chaco\.edm\envs\chaco-test-3.6-pyqt\lib\contextlib.py", line 88, in __exit__
    next(self.gen)
  File "D:\a\chaco\chaco\.edm\envs\chaco-test-3.6-pyqt\lib\site-packages\traitsui\testing\_exception_handling.py", line 74, in reraise_exceptions
    raise RuntimeError(msg)
RuntimeError: Uncaught exceptions found.
=== Exception (type: <class 'AttributeError'>, value: '_QtWindow' object has no attribute 'handler') ===
Traceback (most recent call last):
  File "D:\a\chaco\chaco\.edm\envs\chaco-test-3.6-pyqt\lib\site-packages\enable\qt4\base_window.py", line 258, in leaveEvent
    self.handler.leaveEvent(event)
AttributeError: '_QtWindow' object has no attribute 'handler'

----------------------------------------------------------------------
```
I had thought it was consistently occurring, so when it didn't happen on #663  (and on any commits since then...) I had assumed that fixed it.

Looks like this is really an intermittent failure and will require more investigation.

I am tempted to just skip the test on windows/pyqt for now until it can truly be resolved so it doesn't interfere with other PRs.  Will do this in a separate PR.

**Pull request merged**